language-support/hs/bindings/src/DA/Ledger/Services/PackageManagementService.hs

```diff
@@ -3,25 +3,66 @@
 
 {-# LANGUAGE GADTs #-}
 
-module DA.Ledger.Services.PackageManagementService (uploadDarFile) where
+module DA.Ledger.Services.PackageManagementService (
+    listKnownPackages, PackageDetails(..),
+    uploadDarFile,
+    ) where
 
-import Com.Digitalasset.Ledger.Api.V1.Admin.PackageManagementService
-import Data.ByteString(ByteString)
+import DA.Ledger.Convert
 import DA.Ledger.GrpcWrapUtils
 import DA.Ledger.LedgerService
+import DA.Ledger.Types
+import Data.ByteString(ByteString)
+import Data.Text.Lazy (Text)
 import Network.GRPC.HighLevel.Generated
+import qualified Com.Digitalasset.Ledger.Api.V1.Admin.PackageManagementService as LL
+
+data PackageDetails = PackageDetails
+    { pid :: PackageId
+    , size :: Int
+    , knownSince :: Timestamp
+    , sourceDescription :: Text
+    } deriving (Eq,Ord,Show)
+
+listKnownPackages :: LedgerService [PackageDetails]
+listKnownPackages =
+    makeLedgerService $ \timeout config ->
+    withGRPCClient config $ \client -> do
+        service <- LL.packageManagementServiceClient client
+        let LL.PackageManagementService {packageManagementServiceListKnownPackages=rpc} = service
+        let request = LL.ListKnownPackagesRequest
+        rpc (ClientNormalRequest request timeout emptyMdm)
+            >>= \case
+            ClientNormalResponse response _m1 _m2 _status _details -> do
+                either (fail . show) return $ raiseResponse response
+            ClientErrorResponse e -> do
+                fail (show e)
+
+raiseResponse ::  LL.ListKnownPackagesResponse -> Perhaps [PackageDetails]
+raiseResponse = \case
```

Comment:
```suggestion
raiseResponse = raiseList raisePackageDetails . LL.listKnownPackagesResponsePackageDetails
```
and remove the next two lines.

Reply:
I prefer to have all the conversion code (most of it it is `Convert.hs`) to follow a common style.
I like the `\case` syntax, because it's the same whether the types have 1 or more variant.
I don't see anything is gained by using a different syntax for the case of 1-variant